scripts/azureml-assets/azureml/assets/release/asset_publish.py

```diff
@@ -40,18 +46,126 @@ def test_files_preprocess(test_jobs, asset_ids: dict):
         logger.print(f"processing test job: {test_job}")
         with open(test_job) as fp:
             data = yaml.load(fp, Loader=yaml.FullLoader)
-            for job_name, job in data["jobs"].items():
-                asset_name = job["component"]
+            for job_name, job in data['jobs'].items():
+                asset_name = job['component']
                 logger.print(f"processing asset {asset_name}")
                 if asset_name in asset_ids:
-                    job["component"] = asset_ids.get(asset_name)
-                    logger.print(f"for job {job_name}, the new asset id: {job['component']}")
+                    job['component'] = asset_ids.get(asset_name)
+                    logger.print(
+                        f"for job {job_name}, the new asset id: {job['component']}")
             with open(test_job, "w") as file:
-                yaml.dump(
-                    data,
-                    file,
-                    default_flow_style=False,
-                    sort_keys=False)
+                yaml.dump(data, file, default_flow_style=False,
+                          sort_keys=False)
+
+
+def update_spec_file(spec_file: Path, path: Path):
+    """Update the yaml file after getting the model has been prepared."""
+    with open(spec_file) as f:
+        model_file = yaml.safe_load(f)
+    model_file['path'] = path
+    with open(spec_file, "w") as f:
+        yaml.dump(model_file, f)
+
+
+def model_prepare(
+    model_config: assets.ModelConfig, spec_file: Path, model_dir: Path
+) -> Dict[str, Path]:
```

Comment:
`Path` wouldn't apply to the `download_success` boolean. Also consider raising exceptions when things break instead of passing booleans around.
```suggestion
) -> Dict[str, object]:
```